src/amr_criteria/refinement_package.cpp

```diff
@@ -163,10 +278,13 @@ TaskStatus Tag(MeshBlockData<Real> *rc) {
 }
 
 template <>
-TaskStatus Tag(MeshData<Real> *rc) {
+TaskStatus Tag(MeshData<Real> *md) {
   PARTHENON_INSTRUMENT
-  for (int i = 0; i < rc->NumBlocks(); i++) {
-    SetRefinement_(rc->GetBlockData(i).get());
+  ParArray1D<AmrTag> amr_tags = CheckAllRefinement(md);
+  auto amr_tags_h = amr_tags.GetHostMirrorAndCopy();
+
+  for (int i = 0; i < md->NumBlocks(); i++) {
+    SetRefinement_(md->GetBlockData(i).get(), amr_tags_h(i));
```

Comment:
So just to double check that I fully follow the logic (here and above).
Originally, we called the block version that contained both the internal refinement conditions as well as the ones enrolled in the packages.
In the new version, the internal ones are now `MeshData`, so that's why the new `CheckAllRefinement` is called which contains both the internal ones as well as the package `MeshData` ones.
In addition, we also all the block version (which now only calls the package block version but no internal checks any more).
Correct?

Reply:
yes that is all correct